### PR TITLE
Fix bugs; address adngdb's comments

### PR DESCRIPTION
**Pull request by willkg**
* flesh out some of the examples and reduce clutter
* clean up msg building code
* adjust configuration docs--hopefully this is a better and clearer flow

### CODE REVIEW

File: docs/configuration.rst [hunk 1]
```diff
@@ -19,16 +22,61 @@ For example:
...
+     File "configuration_doc.py", line 22, in <module>
+       main()
+     File "configuration_doc.py", line 12, in main
+       doc='True to put the app in debug mode. Don\'t use this in production!'
+     File "/home/willkg/mozilla/everett/everett/manager.py", line 936, in __call__
+       raise InvalidValueError(msg)
+   everett.InvalidValueError: ValueError: "omg" is not a valid bool value
+   namespace=None key=debug requires a value parseable by everett.manager.parse_bool
+   True to put the app in debug mode. Don't use this in production!
+   For configuration help, see https://example.com/configuration
```
Comment (willkg) [inline docs/configuration.rst]: @adngdb Does this section look better?

File: everett/manager.py [hunk 2]
```diff
@@ -912,24 +929,19 @@ def __call__(self, key, namespace=None, default=NO_VALUE,
...
+                                'msg': str(exc_info[1])
+                            },
+                            'namespace=%(namespace)s key=%(key)s requires a value parseable by %(parser)s' % {
+                                'namespace': use_namespace,
+                                'key': key,
+                                'parser': qualname(parser)
+                            },
+                            doc,
+                            self.doc
+                        )
```
Comment (willkg) [inline everett/manager.py]: @adngdb Is this easier to read?

Comment (adngdb) [inline everett/manager.py]: Wow yes, I especially like your cleverness with that ``build_msg`` function!

File: docs/configuration.rst (hunk 1 above)
Comment (adngdb) [inline docs/configuration.rst]: :+1: this is great!

**Comment by adngdb** [comment]
I really like what you have done here!

**Comment by willkg** [comment]
@adngdb Thank you!

**Pull request merged**